### PR TITLE
Feature - Add Subplan to Program Plan

### PR DESCRIPTION
**Pull request by aidengeorge97**
Addresses #199.

This PR will allow students to nominate which subplan they wish to complete, given the subplan rules of their program. 

image

Given the acceptance criteria of #199, this does not close the issue as it does not "add the requirements of that subplan to the program plan". It does complete the first criterion however, and may be appropriate for this prototype.

This feature was implemented before #235 was pulled. I believe there is room for cleaner integration of the two features in both UI and code, but in its current state this feature is functional.

### CODE REVIEW

**Review by JackSekoranja — APPROVED** [review]
Works as advertised. Happy for you to merge once the pep-8 issues are fixed

File: cassdegrees/static/js/student_edit.js [hunk 1]
```diff
@@ -164,3 +164,29 @@ interact('.dropzone').dropzone({
...
+        div = div.nextSibling;
+    }
+    if (div.style.display === "none") {
+        div.style.display = "block";
+        element.innerHTML = "Hide Options";
+    } else {
+        div.style.display = "none";
+        element.innerHTML = "Show Options";
+    }
+}
```
Comment (Daniel-Jang) [inline cassdegrees/static/js/student_edit.js]: New line

File: cassdegrees/templates/widgets/subplan_choice.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    <div class="card selectable-card" onclick="subplanSelect(this)" name='{{ subplan.code }} - {{ subplan.name }}'>
+                        <div class="box-solid">
+                            <b class="course-title">{{ subplan.code }} - {{ subplan.name }}</b>
+                        </div>
+                    </div>
+                {% endif %}
+            {% endfor %}
+        </div>
+    </div>
+</div>
```
Comment (Daniel-Jang) [inline cassdegrees/templates/widgets/subplan_choice.html]: New line needed here

**Review by Daniel-Jang — APPROVED** [review]
I had a go as well and it seemed to work well. Once you add the new lines to the files I have indicated, I'm happy for you to merge.

**Comment by Daniel-Jang** [comment]
Uh oh, I just tried to save the plan with a subplan selected and got this:
image

Do you know anything about this?

Also, you mentioned that you only meet the first acceptance criteria of #199, are you counting the saving bit as part of the second criteria?

**Review by Daniel-Jang — CHANGES_REQUESTED** [review]
Just marking as "request changes" for now, until I understand the situation with the crashing.

**Comment by aidengeorge97** [comment]
> Uh oh, I just tried to save the plan with a subplan selected and got this:
> Do you know anything about this?

Hmm I'll investigate it now.

> Also, you mentioned that you only meet the first acceptance criteria of #199, are you counting the saving bit as part of the second criteria?

For the second criterion I interpret it as suggesting which courses/rules are required by including the subplan choice in the program. I would consider the ability to save the subplan as part of the first criterion.

**Comment by aidengeorge97** [comment]
> Just marking as "request changes" for now, until I understand the situation with the crashing.

I believe I've solved the issue now, thanks for pointing it out.

**Comment by Daniel-Jang** [comment]
> > Just marking as "request changes" for now, until I understand the situation with the crashing.
> 
> I believe I've solved the issue now, thanks for pointing it out.

Saving now works, will approve. No worries.

**Review by Daniel-Jang — APPROVED** [review]

**Pull request merged**